### PR TITLE
Add handler for single-element (int) arrays without brackets on CLI

### PR DESCRIPTION
**Pull request by Mierdin**
> Due to some CI issues, I am closing this PR and re-opening as a PR from the main repo, instead of my fork: https://github.com/StackStorm/st2/pull/3161

If a user wants to use an action with an array parameter, and passes a single int to this parameter thusly:

	st2 run foopack.fooaction arrayparam=51

the API throws an error because the client renders this as an int, and not an array of length 1.

This was because the transform_array function in actions.py first tries to
load the given value as JSON. In the example above, 51 is a valid int in JSON,
so this results in an error because of a mismatch between what the API expected
(array) and what was actually sent (int).

Note that this isn't a problem if the single element in this array is a string; in that case, the exception causes the function to split on commas, and since there aren't any, it returns a list containing a single string member. However, if it's an int, for some reason, json.loads works fine and this bug appears.

To get around this, the user **could** run something like:

	st2 run foopack.fooaction arrayparam=[51]

However, this is a bit unintuitive. Because of the action metadata, we already know
that this **should** be a list, so we can force it to be. So, this commit adds a
step to check if the value is an int, and if so, force it to be a Python
list. This allows the user to also use the more intutive former example.

Closes https://github.com/StackStorm/st2/issues/3157

### CODE REVIEW

**Comment by Kami** [comment]
LGTM :+1:

File: st2client/st2client/commands/action.py [hunk 1]
```diff
@@ -503,6 +503,26 @@ def transform_object(value):
...
+            #
+            # Normally, json.loads can handle this, but not if the single element is
+            # an int.
+            #
+            # We want to try to handle this first, so it doesn't get accidentally
+            # sent to the API as an int, instead of an array of single-element int.
+            try:
+                # Force this to be a list containing the single int, then
+                # cast the whole thing to string so json.loads can handle it
+                value = str([int(value)])
```
Comment (Kami) [inline st2client/st2client/commands/action.py]: Another way we could improve handling of the array parameters in the future -

Users can also specify type for each array parameter inside an action metadata (`items: type: int` for example) and this information is available here so we could also leverage that to cast each array item to the correct type.

If nothing else we should probably just open a feature request issue for that change :)

Comment (Mierdin) [inline st2client/st2client/commands/action.py]: That's a good idea! I opened https://github.com/StackStorm/st2/issues/3160 to track this.

**Comment by Mierdin** [comment]
Due to some CI issues, I am closing this PR and re-opening as a PR from the main repo, instead of my fork: https://github.com/StackStorm/st2/pull/3161

**Pull request closed without merging**